runtime/queries/luau/highlights.scm

```diff
@@ -0,0 +1,794 @@
+"return" @keyword.control.return
+
+"local" @keyword.storage.modifier
+
+"type" @keyword.storage.type
+
+"function" @keyword.function
+
+(break_stmt) @keyword.control
+(continue_stmt) @keyword.control
+(readwrite) @keyword.storage.modifier
+
+[
+  "do"
+  "end"
+] @keyword
+
+[
+  "while"
+  "repeat"
+  "until"
+  "for"
+] @keyword.control.repeat
+
+[
+  "if"
+  "elseif"
+  "else"
+  "then"
+] @keyword.control.conditional
+
+[
+  "in"
+  "and"
+  "or"
+  "not"
+] @keyword.operator
+
+(ifexp
+[
+  "if"
+  "then"
+  "elseif"
+  "else"
+] @keyword.operator)
+
+(type_stmt "export" @keyword.control.import) 
+
+(_
+  operator: [
+    "+" "-" "*"  "/"  "//" "%"
+    "^" "#" "==" "~=" "<=" ">="
+    "<" ">" "&"  "|"  "->" "::"
+    ".."
+  ] @operator
+)
+
+(_
+  assign_symbol: [
+    "="  "+=" "-=" "*=" "/=" "//="
+    "%=" "^=" "..="
+  ] @operator
+)
+
+[
+  ";"
+  ":"
+  ","
+  "."
+] @punctuation.delimiter
+
+(string) @string
+
+(_
+  variable_name: (name) @variable
+)
+
+(_
+  parameter_name: (name) @variable.parameter
+)
+
+(_
+  method_name: (name) @function.method
+)
+
+(_
+  function_name: (name) @function
+)
+
+(_
+  table_name: (name) @namespace
+)
+
+(_
+  field_name: (name) @variable.other.member
+)
+
+(ERROR) @error
```

Comment:
```suggestion
```

We typically do not highlight syntax errors as it's quite noisy while typing. We might add a config option for it in the future though.